packages/dbml-core/src/parse/dbml/src/lib/result.ts

```diff
@@ -0,0 +1,34 @@
+import { ParsingError } from './errors';
+
+export default class Result<T> {
+  private value?: T;
+
+  private errors: ParsingError[];
```

Comment:
If it could be used for other generic use cases, does it make sense to hardcode the error type for this?